scalableminds/webknossos
# Bucket loading meter

This PR adds the current average download speed of buckets to the statusbar.

### URL of deployed dev instance (used for testing):
- https://bucketloadingmeter.webknossos.xyz/

### Steps to test:
- Open a dataset in view mode or any tracing.
- While the buckets are being loaded the download indicator in the statusbar should show the current download speed. Note that the past downloaded buckets of up to 5 seconds are considered for the calculation. This means that up to five seconds long after nothing is downloaded, the status indication still shows that buckets are being downloaded.
- Hover the download indication in the status bar. A tooltip should show the accumulated downloaded data amount in the form of buckets. To test this do the following:
  - Wait until no buckets are being downloaded anymore. The download speed should be 0.
  - Open the dev tools, open the network tab and start recording. Write down the currently accumulated amount of downloaded bucket data from the tooltip.
  - Reload a visible layer in the layer settings by clicking on the reload icon. The network tab should now list the requests of the layer data being prefetched. 
  - Once the download speed is at 0 again the reloading of the layer should be complete. Below the list, in the network tab, there is the data amount of resources. Calculate the difference between the written down value prior to the layer reloading and what is now shown in the download speed tooltip as the total amount. The difference should match about the same amount as what is given by the network tab.

- I do not know how to test, whether the shown download speed is actually accurate 🤔 ....

### Issues:
- No issue exists for this pr

------
(Please delete unneeded items, merge only when none are left open)
